transport/http/handler.go

```diff
@@ -144,8 +143,12 @@ func handleOnewayRequest(
 	}
 	treq.Body = &buff
 
+	// create a new context for oneway requests since the HTTP handler cancels
+	// http.Request's context when ServeHTTP returns
+	ctx := opentracing.ContextWithSpan(context.Background(), span)
```

Comment:
Oneway transports can make any level of guarantees about “delivery”. On the UDP side, they can drop failure on the floor. On the SQS side, they can guarantee that the messages will eventually be processed without an error. We need to be clear that, with the implementation as is, we have erred on the side of UDP. Regardless, the oneway handler must have a deadline, but it is up to the handler and disjoint from the caller’s TTL (which applies only to time to ack).